ctapipe/io/hessiofilereader.py

```diff
@@ -0,0 +1,211 @@
+        except ImportError:
+            msg = "The `pyhessio` python module is required to access MC data"
+            self.log.error(msg)
+            raise
+
+        self.open_hessio = open_hessio
+        self.close_hessio = close_file
+        self.HessioGeneralError = HessioGeneralError
+
+        self.allowed_tels = None
+
+        if HessioFileReader._count > 0:
+            self.log.warn("Only one pyhessio reader allowed at a time. "
+                          "Previous hessio file will be closed.")
+            self.close_hessio()
+        HessioFileReader._count += 1
+
+        self._metadata['is_simulation'] = True
+
+    @staticmethod
+    def is_compatible(file_path):
+        return file_path.endswith('.gz')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close_hessio()
+
+    def _generator(self):
+        with self.open_hessio(self.input_path) as pyhessio:
+            # the container is initialized once, and data is replaced within
+            # it after each yield
+            Provenance().add_input_file(self.input_path, role='dl0.sub.evt')
+            counter = 0
+            eventstream = pyhessio.move_to_next_event()
+            if self.allowed_tels is not None:
+                self.allowed_tels = set(self.allowed_tels)
+            data = DataContainer()
+            data.meta['origin'] = "hessio"
+
+            # some hessio_event_source specific parameters
+            data.meta['input'] = self.input_path
+            data.meta['max_events'] = self.max_events
+
+            for event_id in eventstream:
+
+                if counter == 0:
+                    # subarray info is only available when an event is loaded,
+                    # so load it on the first event.
+                    data.inst.subarray = self._build_subarray_info(pyhessio)
+
+                run_id = pyhessio.get_run_number()
+                tels_with_data = set(pyhessio.get_teldata_list())
+                data.count = counter
+                data.r0.run_id = run_id
+                data.r0.event_id = event_id
+                data.r0.tels_with_data = tels_with_data
+                data.r1.run_id = run_id
+                data.r1.event_id = event_id
+                data.r1.tels_with_data = tels_with_data
+                data.dl0.run_id = run_id
+                data.dl0.event_id = event_id
+                data.dl0.tels_with_data = tels_with_data
+
+                # handle telescope filtering by taking the intersection of
+                # tels_with_data and allowed_tels
+                if self.allowed_tels is not None:
+                    selected = tels_with_data & self.allowed_tels
+                    if len(selected) == 0:
+                        continue  # skip event
+                    data.r0.tels_with_data = selected
+                    data.r1.tels_with_data = selected
+                    data.dl0.tels_with_data = selected
+
+                data.trig.tels_with_trigger \
```

Comment:
backslash line continuation is possibly dangerous, when some poor person adds a single whitespace. 

https://www.python.org/dev/peps/pep-0008/#maximum-line-length
> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

Reply:
Ok